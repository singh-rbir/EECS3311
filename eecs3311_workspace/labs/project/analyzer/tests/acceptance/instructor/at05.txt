add_class("A")
add_class("B")

-- Error: Duplicated parameter names.

-- Add command `ca(int x, boolean y, A x, B y)` to class `A`.
-- Report in the order in which the duplicates are found.
-- Here the 3rd and 4th parameters are duplicates
add_command ("A", "ca", <<["x", "INTEGER"], ["y", "BOOLEAN"], ["x", "A"], ["y", "B"]>>)
-- Add query `cq(B x, int y, A y, boolean x)` to class `A`.
-- Here the 3rd and 4th parameters are duplicates.
add_command ("A", "cq", <<["x", "B"], ["y", "INTEGER"], ["y", "A"], ["x", "BOOLEAN"]>>)

