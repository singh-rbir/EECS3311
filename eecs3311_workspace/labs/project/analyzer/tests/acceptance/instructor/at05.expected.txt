  Status: OK.
  Number of classes being specified: 0
->add_class("A")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_class("B")
  Status: OK.
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_command("A","ca",<<["x", "INTEGER"], ["y", "BOOLEAN"], ["x", "A"], ["y", "B"]>>)
  Status: Error (Duplicated parameter names: x, y).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_command("A","cq",<<["x", "B"], ["y", "INTEGER"], ["y", "A"], ["x", "BOOLEAN"]>>)
  Status: Error (Duplicated parameter names: y, x).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
